src/backend/distributed/executor/multi_utility.c

```diff
@@ -1255,7 +1255,13 @@ ExecuteDistributedDDLCommand(Oid relationId, const char *ddlCommandString,
 {
 	bool executionOK = false;
 
-	PreventTransactionChain(isTopLevel, "distributed DDL commands");
+	if (IsInTransactionChain(isTopLevel) && XactOperations != 0)
```

Comment:
I suppose the conditions under which the message could be emitted are the same with or without the `IsInTransactionChain`, but without it, we'd need a comment for a reader. I can change it to a comment if you want.